parcels/kernel.py

```diff
@@ -96,24 +99,51 @@ def __init__(self, fieldset, ptype, pyfunc=None, funcname=None,
 
             basename = path.join(get_cache_dir(), self._cache_key)
             self.src_file = "%s.c" % basename
-            self.lib_file = "%s.so" % basename
+            self.lib_file = "%s.%s" % (basename, 'dll' if platform == 'win32' else 'so')
             self.log_file = "%s.log" % basename
         self._lib = None
 
+    def __del__(self):
+        # Clean-up the in-memory dynamic linked libraries.
+        # This is not really necessary, as these programs are not that large, but with the new random
+        # naming scheme which is required on Windows OS'es to deal with updates to a Parcels' kernel.
+        if not self._lib == None:
+            _ctypes.FreeLibrary(self._lib._handle) if platform == 'win32' else _ctypes.dlclose(self._lib._handle)
+            del self._lib
+            self._lib = None
+            map(remove, [self.src_file, self.lib_file, self.log_file]) if path.isfile(self.lib_file) else None
+
+
     @property
     def _cache_key(self):
         field_keys = "-".join(["%s:%s" % (name, field.units.__class__.__name__)
                                for name, field in self.field_args.items()])
-        key = self.name + self.ptype._cache_key + field_keys
+        key = self.name + self.ptype._cache_key + field_keys + ('TIME:%f' % time.time())
         return md5(key.encode('utf-8')).hexdigest()
 
+    def remove_lib(self):
+        # Unload the currently loaded dynamic linked library to be secure
+        if not self._lib == None:
+            _ctypes.FreeLibrary(self._lib._handle) if platform == 'win32' else _ctypes.dlclose(elf._lib._handle)
```

Comment:
Type here? Do you mean `_ctypes.dlclose(self._lib._handle)` (so with `self` instead of `elf`)?